test/test_images/runtime/handlers/file_access_attempt.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package handlers
+
+import (
+	"io/ioutil"
+	"os"
+
+	"knative.dev/serving/test/types"
+)
+
+type permissionBits uint32
+
+const (
+	otherRead permissionBits = 1 << (2 - iota)
+	otherWrite
+)
+
+func (p permissionBits) hasPermission(mode permissionBits) bool {
+	return p&mode == mode
+}
+
+func fileAccessAttempt(filePaths ...string) map[string]types.FileAccessInfo {
+	files := map[string]types.FileAccessInfo{}
+	for _, path := range filePaths {
+		accessInfo := types.FileAccessInfo{}
+
+		if err := checkReadable(path); err != nil {
+			accessInfo.ReadErr = err.Error()
+		}
+
+		if err := checkWritable(path); err != nil {
+			accessInfo.WriteErr = err.Error()
+		}
+
+		files[path] = accessInfo
+	}
+	return files
+}
+
+// checkReadable function checks whether path file or directory is readable
+func checkReadable(path string) error {
+	file, err := os.Stat(path) // It should only return error only if file does not exist
+	if err != nil {
+		return err
+	}
+
+	// We aren't expected to be able to read, so just exit
+	perm := permissionBits(file.Mode().Perm())
+	if !perm.hasPermission(otherRead) {
+		return nil
+	}
+
+	if file.IsDir() {
+		_, err := ioutil.ReadDir(path)
+		return err
+	}
+	readFile, err := os.OpenFile(path, os.O_RDONLY, 0)
+	defer readFile.Close()
```

Comment:
I am close to be sure that this will panic, if the error is not nil (since if file isn't there it won't open it).
```
if err != nil {return err}
readFile.Close()
return nil
```